tensorboard/plugins/debugger/debugger_plugin_loader.py

```diff
@@ -91,42 +125,46 @@ def load(self, context):
-          '  pip install tensorflow')
-    try:
+    if flags.debugger_data_server_grpc_port > 0 or flags.debugger_port > 0:
+      # Verify that the required Python packages are installed.
+      try:
+        # pylint: disable=g-import-not-at-top,unused-import
+        import tensorflow
+      except ImportError:
+        raise ImportError(
+            'To use the debugger plugin, you need to have TensorFlow installed:\n'
+            '  pip install tensorflow')
+
+    if flags.debugger_data_server_grpc_port > 0:
       # pylint: disable=line-too-long,g-import-not-at-top
       from tensorboard.plugins.debugger import debugger_plugin as debugger_plugin_lib
-      from tensorboard.plugins.debugger import interactive_debugger_plugin as interactive_debugger_plugin_lib
       # pylint: enable=line-too-long,g-import-not-at-top
-    except ImportError as e:
-      e_type, e_value, e_traceback = sys.exc_info()
-      message = e.msg if hasattr(e, 'msg') else e.message  # Handle py2 vs py3
-      if 'grpc' in message:
-        e_value = ImportError(
-            message +
-            '\n\nTo use the debugger plugin, you need to have '
-            'gRPC installed:\n  pip install grpcio')
-      six.reraise(e_type, e_value, e_traceback)
-    if flags.debugger_port > 0:
-      interactive_plugin = (
-          interactive_debugger_plugin_lib.InteractiveDebuggerPlugin(context))
-      logger.info('Starting Interactive Debugger Plugin at gRPC port %d',
-                   flags.debugger_data_server_grpc_port)
-      interactive_plugin.listen(flags.debugger_port)
-      return interactive_plugin
-    elif flags.debugger_data_server_grpc_port > 0:
+
+      # debugger_data_server_grpc opens the non-interactive Debugger Plugin,
+      # which appears as health pills in the Graph Plugin.
       noninteractive_plugin = debugger_plugin_lib.DebuggerPlugin(context)
       logger.info('Starting Non-interactive Debugger Plugin at gRPC port %d',
                    flags.debugger_data_server_grpc_port)
       noninteractive_plugin.listen(flags.debugger_data_server_grpc_port)
       return noninteractive_plugin
-    raise AssertionError()
+    elif flags.debugger_port > 0:
+      # pylint: disable=line-too-long,g-import-not-at-top
+      from tensorboard.plugins.debugger import interactive_debugger_plugin as interactive_debugger_plugin_lib
+      # pylint: enable=line-too-long,g-import-not-at-top
+
+      # If debugger_data_server_grpc_port flag is not specified, we always
+      # instantiate the interactive Debugger Plugin. If debugger_port
+      # is not specified (i.e., defaults to -1), the frontend will display a
+      # message indicating that the plugin is not active. It'll also display
+      # a command snippet to illustrate how to activate the interactive Debugger
+      # Plugin.
+      interactive_plugin = (
+          interactive_debugger_plugin_lib.InteractiveDebuggerPlugin(context))
+      if flags.debugger_port > 0:
```

Comment:
seems like this if statement is not needed because of L151.

Reply:
Good catch. Fixed.